### ISSUE TITLE
Add support for ASYNC

### ISSUE THREAD

**Issue by Lykos153** [Lykos153/AnnexRemote#22]
https://git-annex.branchable.com/design/external_special_remote_protocol/async_appendix/

**Comment by yarikoptic** [comment]
Hi, I wonder if implementing support for ASYNC is on the roadmap?  Asking because I know that @mih et al would be interested as well, and it might motivate me to look into reimplementing our elderly customly coded custom remotes in datalad to just use annexremote as well (instead of just implementing async support in there without RF).

**Comment by Lykos153** [comment]
Yes, that's the next thing I'm going to spend my time on after fixing [this google issue](https://github.com/Lykos153/git-annex-remote-googledrive/issues/52)!
